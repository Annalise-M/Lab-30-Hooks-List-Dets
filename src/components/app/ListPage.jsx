/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';
import { useState, useEffect } from 'react';
import { getApiAll } from '../../services/heyArnold-api';
import CharacterDetail from './DetailsPage';

const ListPage = () => {
  const [character, setCharacter] = useState([]);

  useEffect(() => {
    getApiAll()
      .then(character => setCharacter(character));
  }, []);
  
  return (
    <div>
      <h3>Character List</h3>
      {character.map((character) => {
        return (
          <li key={character._id}>
            <CharacterDetail {...character}/>
         
            <br />
          </li>
        );
      })}
    </div>

  );
};

ListPage.propTypes = {

  characters: PropTypes.array
 
};

export default ListPage;
