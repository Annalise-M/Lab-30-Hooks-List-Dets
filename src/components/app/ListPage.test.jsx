import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ListPage from './ListPage';

describe('ListPage container', () => {
  afterEach(() => cleanup());
  it('displays HeyArnold character data', () => {
    const { asFragment } = render(<ListPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
