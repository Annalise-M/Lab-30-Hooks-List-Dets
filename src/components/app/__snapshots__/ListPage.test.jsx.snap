// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListPage container displays HeyArnold character data 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Character List
    </h3>
  </div>
</DocumentFragment>
`;
